pkg/cpio/fs_unix.go

```diff
@@ -159,12 +159,12 @@ func CreateFileInRoot(f Record, rootDir string, forcePriv bool) error {
 		}
 
 	case os.ModeDevice:
-		if err := syscall.Mknod(f.Name, perm(f)|syscall.S_IFBLK, dev(f)); err != nil && forcePriv {
+		if err := Mknod(f.Name, perm(f)|syscall.S_IFBLK, dev(f)); err != nil && forcePriv {
 			return err
 		}
 
 	case os.ModeCharDevice:
-		if err := syscall.Mknod(f.Name, perm(f)|syscall.S_IFCHR, dev(f)); err != nil && forcePriv {
+		if err := Mknod(f.Name, perm(f)|syscall.S_IFCHR, dev(f)); err != nil && forcePriv {
```

Comment:
could you try calling unix.Mknod here and see if it removes the need for the mknod_*.go files you added? I don't see the need at present?

Reply:
That didn't work, unfortunately. I've lowercased it. :sparkles:

Reply:
what did not work? If it's just a matter of type of arg, a more common fix is create a per-os or per-architecture  function for the arg that's different.

But just wondering, what does "didn't work" mean in this case?

Reply:
So you mean a per-OS `dev()` wrapper instead lf the per-OS `mknod`?

What didn't work was simply replacing the original `syscall.Mknod` with `unix.Mknod`. I get the same error as before (like https://bugs.freebsd.org/bugzilla/show_bug.cgi?id=236131 ) because FreeBSD wants a `uint64`.